protocols/delegate-factory/contracts/DelegateFactory.sol

```diff
@@ -17,22 +17,34 @@
 pragma solidity 0.5.10;
 
 import "@airswap/delegate/contracts/Delegate.sol";
+import "@airswap/swap/contracts/interfaces/ISwap.sol";
 import "@airswap/indexer/contracts/interfaces/ILocatorWhitelist.sol";
 import "@airswap/delegate-factory/contracts/interfaces/IDelegateFactory.sol";
 
 contract DelegateFactory is IDelegateFactory, ILocatorWhitelist {
 
   mapping(address => bool) internal deployedAddresses;
+  ISwap public swapContract;
 
   /**
-    * @notice Create a new Delegate contract
+    * @dev swapContract is unable to be changed after the factory sets it
     * @param _swapContract address of the swap contract the delegate will deploy with
+    */
+  constructor(ISwap _swapContract) public {
```

Comment:
Does this mean it has to be passed to the constructor as an `ISwap`? Could it be an address and then cast in the constructor body?

Reply:
In the tests I have:
```
before('Deploy Delegate Factory', async () => {
    delegateFactory = await DelegateFactory.new(swapContract)
}
```
Passing it in at the web level should be ok.

As for contract to contract it needs to be passed in as `ISwap`. I changed it because I didn't have a compelling reason as to why the `DelegateFactory` would output `Delegates` that don't conform to `ISwap`

We can change it back to address, but then there's no safety on the type that can be passed.

Reply:
Makes sense, was imagining that if it's an `ISwap` then the contract that is constructing it also needs to import / know about `ISwap`. If it's just an address that's cast in the constructor, the contract constructing it does not need to import `ISwap`

Reply:
that's correct. I thought the explicitness was better for safety to ensure type safety a Swap. We can always change it later however.